File: Problem1-2019-02-16/go/main.go
package main

import "fmt"

func calculate(l []int, k int) bool {
	seen := map[int]bool{}
	for _, n := range l {
		diff := k - n
		if _, ok := seen[diff]; ok {
			return true
		}
		seen[n] = true
	}

	return false
}

func main() {
	k := 17
	l := []int{10, 15, 3, 7}
	fmt.Println(calculate(l, k))
}

File: Problem1-2019-02-16/ruby/main.rb
#!/usr/bin/env ruby

require "set"
require "minitest/autorun"

def calculate(num_list, num_sum)
  seen = Set[]
  num_list.each do |n|
    diff = num_sum - n
    if seen.include?(diff)
      return true
    end
    seen.add(n)
  end
  return false
end

class TestCalculate < Minitest::Test
  def test_calculate
    l = [10, 15, 3, 7]
    k = 17
    r = calculate(l, k)
    assert_equal(r, true)
  end
end
